.github/workflows/carpenter_push.yaml

```diff
@@ -2,24 +2,44 @@ name: Carpentry Push
 on:
   push:
     branches:
-      - main
+      - "**"
 env:
-  IMAGE_NAME: arcaflow-plugin-template-python
+  IMAGE_NAME: arcaflow-plugin-uperf
   IMAGE_TAG: 'latest'
+  QUAY_IMG_EXP: 'never'
   GITHUB_USERNAME: ${{ github.actor }}
   GITHUB_PASSWORD: ${{ secrets.GITHUB_TOKEN }}
   GITHUB_NAMESPACE: ${{ github.repository_owner }}
   QUAY_NAMESPACE: ${{ secrets.QUAY_NAMESPACE }}
   QUAY_USERNAME: ${{ secrets.QUAY_USERNAME }}
   QUAY_PASSWORD: ${{ secrets.QUAY_PASSWORD }}
 jobs:
-  carpenter-push:
-    name: carpenter_push
+  carpenter-build-prod:
+    name: carpenter_build_prod
     runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/main'
     steps:
       - name: Checkout this project
         uses: actions/checkout@v3
       - name: carpenter build
         uses: arcalot/arcaflow-plugin-image-builder@main
         with:
           args: build --build --push
+  carpenter-build-dev:
+    name: carpenter_build_dev
+    runs-on: ubuntu-latest
+    if: github.ref != 'refs/heads/main'
+    steps:
+      - name: set_image_tag
+        run: |
+          export commit_hash=${{ github.sha }}
+          echo "IMAGE_TAG=${GITHUB_REF##*/}_${commit_hash:0:7}" >> $GITHUB_ENV
+      - name: set_quay_image_expiration
+        run: |
+          echo "QUAY_IMG_EXP=90d" >> $GITHUB_ENV
+      - name: Checkout this project
+        uses: actions/checkout@v3
+      - name: carpenter build
+        uses: arcalot/arcaflow-plugin-image-builder@main
+        with:
+          args: build --build --push
```

Comment:
Is the only difference between `carpenter_build.yaml` and `carpenter_push.yaml` the addition of `--push` on lines 27 and 45?

If so, isn't there a way to have the `carpenter_push.yaml` just use `carpenter_build.yaml` telling it to also use `--push`?  That would DRY it out.

Reply:
Tracking in https://github.com/arcalot/arcaflow-plugin-template-python/issues/16.